In [1]:
print("Welcome")

Welcome


**Library**

In [5]:
import pandas as pd
from pandas.io.json import json_normalize  # tranform JSON file into a pandas dataframe

import numpy as np

from bs4 import BeautifulSoup

# Matplotlib and associated plotting modules
import matplotlib.pyplot as plt
import matplotlib.cm as cm
import matplotlib.colors as colors

import seaborn as sns
from urllib.request import urlopen

import requests
# import k-means from clustering stage
from sklearn.cluster import KMeans

print("Library imported 56.")


/usr/local/lib/python3.6/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


Library imported 56.


In [6]:
import folium # map rendering library
from geopy.geocoders import Nominatim # convert an address into latitude and longitude values
print("Library imported 58.")


Library imported 58.


In [7]:
link = "https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M"
df = pd.read_html(link, header = 0)[0]
df.head()

Postal code           Borough                Neighborhood
0         M1A      Not assigned                         NaN
1         M2A      Not assigned                         NaN
2         M3A        North York                   Parkwoods
3         M4A        North York            Victoria Village
4         M5A  Downtown Toronto  Regent Park / Harbourfront

In [8]:
df.to_csv (r'D:\Workspace\data\postal_Toronto.csv', index = None, header=True) 
print("Done1")

Done1


### **Let's Explore venues in South Mumbai **

In [9]:
areas = ["Colaba","Apollo Bandar","Fort","Churchgate","Nariman Point"
        ,"Marine Lines","Walkeshwar","Malabar Hill","Kalbadevi"
        ,"Bhuleshwar","Masjid Bandar","Darukhana","Pydhoni","Nagpada","Dongri","Byculla","Grand Road"
        ,"Mazgaon","Mumbai Central","Mahalaxmi","Worli"
]



full_address = []
 
for area in areas:
   
  full_address.append(area + ", Mumbai")

print("full_address",full_address)
print("areas",areas)

full_address ['Colaba, Mumbai', 'Apollo Bandar, Mumbai', 'Fort, Mumbai', 'Churchgate, Mumbai', 'Nariman Point, Mumbai', 'Marine Lines, Mumbai', 'Walkeshwar, Mumbai', 'Malabar Hill, Mumbai', 'Kalbadevi, Mumbai', 'Bhuleshwar, Mumbai', 'Masjid Bandar, Mumbai', 'Darukhana, Mumbai', 'Pydhoni, Mumbai', 'Nagpada, Mumbai', 'Dongri, Mumbai', 'Byculla, Mumbai', 'Grand Road, Mumbai', 'Mazgaon, Mumbai', 'Mumbai Central, Mumbai', 'Mahalaxmi, Mumbai', 'Worli, Mumbai']
areas ['Colaba', 'Apollo Bandar', 'Fort', 'Churchgate', 'Nariman Point', 'Marine Lines', 'Walkeshwar', 'Malabar Hill', 'Kalbadevi', 'Bhuleshwar', 'Masjid Bandar', 'Darukhana', 'Pydhoni', 'Nagpada', 'Dongri', 'Byculla', 'Grand Road', 'Mazgaon', 'Mumbai Central', 'Mahalaxmi', 'Worli']


In [10]:

dic1 = {"Neighborhood": areas,"Address":full_address}
dic1


{'Address': ['Colaba, Mumbai',
  'Apollo Bandar, Mumbai',
  'Fort, Mumbai',
  'Churchgate, Mumbai',
  'Nariman Point, Mumbai',
  'Marine Lines, Mumbai',
  'Walkeshwar, Mumbai',
  'Malabar Hill, Mumbai',
  'Kalbadevi, Mumbai',
  'Bhuleshwar, Mumbai',
  'Masjid Bandar, Mumbai',
  'Darukhana, Mumbai',
  'Pydhoni, Mumbai',
  'Nagpada, Mumbai',
  'Dongri, Mumbai',
  'Byculla, Mumbai',
  'Grand Road, Mumbai',
  'Mazgaon, Mumbai',
  'Mumbai Central, Mumbai',
  'Mahalaxmi, Mumbai',
  'Worli, Mumbai'],
 'Neighborhood': ['Colaba',
  'Apollo Bandar',
  'Fort',
  'Churchgate',
  'Nariman Point',
  'Marine Lines',
  'Walkeshwar',
  'Malabar Hill',
  'Kalbadevi',
  'Bhuleshwar',
  'Masjid Bandar',
  'Darukhana',
  'Pydhoni',
  'Nagpada',
  'Dongri',
  'Byculla',
  'Grand Road',
  'Mazgaon',
  'Mumbai Central',
  'Mahalaxmi',
  'Worli']}

In [11]:
df2 = pd.DataFrame.from_dict(dic1)
df2

Neighborhood                 Address
0           Colaba          Colaba, Mumbai
1    Apollo Bandar   Apollo Bandar, Mumbai
2             Fort            Fort, Mumbai
3       Churchgate      Churchgate, Mumbai
4    Nariman Point   Nariman Point, Mumbai
5     Marine Lines    Marine Lines, Mumbai
6       Walkeshwar      Walkeshwar, Mumbai
7     Malabar Hill    Malabar Hill, Mumbai
8        Kalbadevi       Kalbadevi, Mumbai
9       Bhuleshwar      Bhuleshwar, Mumbai
10   Masjid Bandar   Masjid Bandar, Mumbai
11       Darukhana       Darukhana, Mumbai
12         Pydhoni         Pydhoni, Mumbai
13         Nagpada         Nagpada, Mumbai
14          Dongri          Dongri, Mumbai
15         Byculla         Byculla, Mumbai
16      Grand Road      Grand Road, Mumbai
17         Mazgaon         Mazgaon, Mumbai
18  Mumbai Central  Mumbai Central, Mumbai
19       Mahalaxmi       Mahalaxmi, Mumbai
20           Worli           Worli, Mumbai

In [12]:
for  neighborhood,address in zip(df2['Neighborhood'], df2['Address']):
     print("neighborhood:", neighborhood)
     print("address",address)

neighborhood: Colaba
address Colaba, Mumbai
neighborhood: Apollo Bandar
address Apollo Bandar, Mumbai
neighborhood: Fort
address Fort, Mumbai
neighborhood: Churchgate
address Churchgate, Mumbai
neighborhood: Nariman Point
address Nariman Point, Mumbai
neighborhood: Marine Lines
address Marine Lines, Mumbai
neighborhood: Walkeshwar
address Walkeshwar, Mumbai
neighborhood: Malabar Hill
address Malabar Hill, Mumbai
neighborhood: Kalbadevi
address Kalbadevi, Mumbai
neighborhood: Bhuleshwar
address Bhuleshwar, Mumbai
neighborhood: Masjid Bandar
address Masjid Bandar, Mumbai
neighborhood: Darukhana
address Darukhana, Mumbai
neighborhood: Pydhoni
address Pydhoni, Mumbai
neighborhood: Nagpada
address Nagpada, Mumbai
neighborhood: Dongri
address Dongri, Mumbai
neighborhood: Byculla
address Byculla, Mumbai
neighborhood: Grand Road
address Grand Road, Mumbai
neighborhood: Mazgaon
address Mazgaon, Mumbai
neighborhood: Mumbai Central
address Mumbai Central, Mumbai
neighborhood: Mahalaxmi
address Ma

In [13]:
borough = 'South Mumbai'
data = []
for  neighborhood,address in zip(df2['Neighborhood'], df2['Address']):
  #print("neighborhood:", neighborhood)
  #print("address:", address)
  geolocator = Nominatim(user_agent="ny_explorer")
  location = geolocator.geocode(address)
  latitude = location.latitude
  longitude = location.longitude
  #print("latitude:", latitude)
  #print("longitude:", longitude)
  data.append([neighborhood,borough,latitude,longitude])

col_names =  ['Neighborhood', 'Borough', 'Latitude','Longitude']
neighborhoods = pd.DataFrame(data,columns=col_names)
print("dtypes : ", neighborhoods.dtypes)
print("rows : ", neighborhoods.shape[0])
neighborhoods.head(20)


dtypes :  Neighborhood     object
Borough          object
Latitude        float64
Longitude       float64
dtype: object
rows :  21


Neighborhood       Borough   Latitude  Longitude
0           Colaba  South Mumbai  18.915091  72.825969
1    Apollo Bandar  South Mumbai  18.918375  72.831443
2             Fort  South Mumbai  18.933266  72.834515
3       Churchgate  South Mumbai  18.935957  72.827340
4    Nariman Point  South Mumbai  18.925951  72.823208
5     Marine Lines  South Mumbai  18.945670  72.823781
6       Walkeshwar  South Mumbai  18.955343  72.807947
7     Malabar Hill  South Mumbai  18.958162  72.803366
8        Kalbadevi  South Mumbai  18.949258  72.827938
9       Bhuleshwar  South Mumbai  18.953771  72.827476
10   Masjid Bandar  South Mumbai  18.953599  72.838126
11       Darukhana  South Mumbai  18.977036  72.849077
12         Pydhoni  South Mumbai  18.953994  72.832379
13         Nagpada  South Mumbai  18.968178  72.828601
14          Dongri  South Mumbai  18.962226  72.836795
15         Byculla  South Mumbai  18.976622  72.832794
16      Grand Road  South Mumbai  18.938771  72.835335
17         Mazgaon  South Mumbai  18.968052  72.840012
18  Mumbai Central  South Mumbai  18.969586  72.819315
19       Mahalaxmi  South Mumbai  18.982568  72.824160

In [109]:
neighborhoods.head(20)

Neighborhood       Borough   Latitude  Longitude
0           Colaba  South Mumbai  18.915091  72.825969
1    Apollo Bandar  South Mumbai  18.918375  72.831443
2             Fort  South Mumbai  18.933266  72.834515
3       Churchgate  South Mumbai  18.935957  72.827340
4    Nariman Point  South Mumbai  18.925951  72.823208
5     Marine Lines  South Mumbai  18.945670  72.823781
6       Walkeshwar  South Mumbai  18.955343  72.807947
7     Malabar Hill  South Mumbai  18.958162  72.803366
8        Kalbadevi  South Mumbai  18.949258  72.827938
9       Bhuleshwar  South Mumbai  18.953771  72.827476
10   Masjid Bandar  South Mumbai  18.953599  72.838126
11       Darukhana  South Mumbai  18.977036  72.849077
12         Pydhoni  South Mumbai  18.953994  72.832379
13         Nagpada  South Mumbai  18.968178  72.828601
14          Dongri  South Mumbai  18.962226  72.836795
15         Byculla  South Mumbai  18.976622  72.832794
16      Grand Road  South Mumbai  18.938771  72.835335
17         Mazgaon  South Mumbai  18.968052  72.840012
18  Mumbai Central  South Mumbai  18.969586  72.819315
19       Mahalaxmi  South Mumbai  18.982568  72.824160

In [0]:
neighborhoods.to_csv("Mumbai_locations2.csv")

### Use geopy library to get the latitude and longitude values of New York City.
### In order to define an instance of the geocoder, we need to define a user_agent. We will name our agent ny_explorer, as shown below.

### Create a map of Mumbai with neighborhoods superimposed on top.


In [15]:
map_mumbai = folium.Map(location=[latitude, longitude], zoom_start=14)
map_mumbai

In [16]:

# add markers to map
for lat, lng, borough, neighborhood in zip(neighborhoods['Latitude'], neighborhoods['Longitude'], neighborhoods['Borough'], neighborhoods['Neighborhood']):
    label = '{}, {}'.format(neighborhood, borough)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_mumbai)  
    
map_mumbai

### Define Foursquare Credentials and Version

In [17]:
CLIENT_ID = 'AED1YJAEJO1BSJVJJ5KT2Y2BDGDVPCB21WUXERHTA4B4XXKC' # your Foursquare ID
CLIENT_SECRET = '5WF0PKUAGWG30ASKZSBV12XHC2IDIQLEHRRPU541XI2JVQCH' # your Foursquare Secret
VERSION = '20180605' # Foursquare API version

print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: AED1YJAEJO1BSJVJJ5KT2Y2BDGDVPCB21WUXERHTA4B4XXKC
CLIENT_SECRET:5WF0PKUAGWG30ASKZSBV12XHC2IDIQLEHRRPU541XI2JVQCH


### Let's create a function to repeat the same process to all the neighborhoods in South Mumbai

In [0]:
def getNearbyVenues(names, latitudes, longitudes, radius=500):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Neighborhood', 
                  'Neighborhood Latitude', 
                  'Neighborhood Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    return(nearby_venues)


### Now write the code to run the above function on each neighborhood and create a new dataframe called south_mumbai_venues.

In [22]:
south_mumbai_data = neighborhoods
print("dtypes:",south_mumbai_data.dtypes)
print("shape:",south_mumbai_data.shape)
south_mumbai_data.head(30)


dtypes: Neighborhood     object
Borough          object
Latitude        float64
Longitude       float64
dtype: object
shape: (21, 4)


Neighborhood       Borough   Latitude  Longitude
0           Colaba  South Mumbai  18.915091  72.825969
1    Apollo Bandar  South Mumbai  18.918375  72.831443
2             Fort  South Mumbai  18.933266  72.834515
3       Churchgate  South Mumbai  18.935957  72.827340
4    Nariman Point  South Mumbai  18.925951  72.823208
5     Marine Lines  South Mumbai  18.945670  72.823781
6       Walkeshwar  South Mumbai  18.955343  72.807947
7     Malabar Hill  South Mumbai  18.958162  72.803366
8        Kalbadevi  South Mumbai  18.949258  72.827938
9       Bhuleshwar  South Mumbai  18.953771  72.827476
10   Masjid Bandar  South Mumbai  18.953599  72.838126
11       Darukhana  South Mumbai  18.977036  72.849077
12         Pydhoni  South Mumbai  18.953994  72.832379
13         Nagpada  South Mumbai  18.968178  72.828601
14          Dongri  South Mumbai  18.962226  72.836795
15         Byculla  South Mumbai  18.976622  72.832794
16      Grand Road  South Mumbai  18.938771  72.835335
17         Mazgaon  South Mumbai  18.968052  72.840012
18  Mumbai Central  South Mumbai  18.969586  72.819315
19       Mahalaxmi  South Mumbai  18.982568  72.824160
20           Worli  South Mumbai  19.011696  72.818070

#### Explore within a radius of 500 and limit number of nearby venues to 100.

In [0]:
radius= 500
LIMIT = 100


In [26]:
south_mumbai_venues = getNearbyVenues(names=south_mumbai_data['Neighborhood'],
                                   latitudes=south_mumbai_data['Latitude'],
                                   longitudes=south_mumbai_data['Longitude']
                                  )
print("dtypes:",south_mumbai_venues.dtypes)
print("shape:",south_mumbai_venues.shape)
south_mumbai_venues.head(30)

Colaba
Apollo Bandar
Fort
Churchgate
Nariman Point
Marine Lines
Walkeshwar
Malabar Hill
Kalbadevi
Bhuleshwar
Masjid Bandar
Darukhana
Pydhoni
Nagpada
Dongri
Byculla
Grand Road
Mazgaon
Mumbai Central
Mahalaxmi
Worli
dtypes: Neighborhood               object
Neighborhood Latitude     float64
Neighborhood Longitude    float64
Venue                      object
Venue Latitude            float64
Venue Longitude           float64
Venue Category             object
dtype: object
shape: (359, 7)


Neighborhood  ...             Venue Category
0          Colaba  ...                Men's Store
1          Colaba  ...          German Restaurant
2          Colaba  ...                        Spa
3          Colaba  ...                Coffee Shop
4          Colaba  ...                    Brewery
5          Colaba  ...          Indian Restaurant
6          Colaba  ...                 Restaurant
7          Colaba  ...         Chinese Restaurant
8          Colaba  ...  Middle Eastern Restaurant
9          Colaba  ...                Pizza Place
10         Colaba  ...                       Café
11         Colaba  ...                      Hotel
12         Colaba  ...         Chinese Restaurant
13  Apollo Bandar  ...                      Diner
14  Apollo Bandar  ...          Indian Restaurant
15  Apollo Bandar  ...               Dessert Shop
16  Apollo Bandar  ...                        Bar
17  Apollo Bandar  ...          Indian Restaurant
18  Apollo Bandar  ...                      Hotel
19  Apollo Bandar  ...                     Lounge
20  Apollo Bandar  ...          Indian Restaurant
21  Apollo Bandar  ...         Falafel Restaurant
22  Apollo Bandar  ...                  Juice Bar
23  Apollo Bandar  ...                Flea Market
24  Apollo Bandar  ...                       Café
25  Apollo Bandar  ...                Coffee Shop
26  Apollo Bandar  ...                       Café
27  Apollo Bandar  ...               Cocktail Bar
28  Apollo Bandar  ...           Asian Restaurant
29  Apollo Bandar  ...                      Hotel

[30 rows x 7 columns]

### It returned total 359 venues. Data looks insufficient. Let's increase first Limit to 500.

In [29]:
radius= 500
LIMIT = 500
south_mumbai_venues2 = getNearbyVenues(names=south_mumbai_data['Neighborhood'],
                                   latitudes=south_mumbai_data['Latitude'],
                                   longitudes=south_mumbai_data['Longitude']
                                  )
print("dtypes:",south_mumbai_venues2.dtypes)
print("shape:",south_mumbai_venues2.shape)
south_mumbai_venues2.head(30)

Colaba
Apollo Bandar
Fort
Churchgate
Nariman Point
Marine Lines
Walkeshwar
Malabar Hill
Kalbadevi
Bhuleshwar
Masjid Bandar
Darukhana
Pydhoni
Nagpada
Dongri
Byculla
Grand Road
Mazgaon
Mumbai Central
Mahalaxmi
Worli
dtypes: Neighborhood               object
Neighborhood Latitude     float64
Neighborhood Longitude    float64
Venue                      object
Venue Latitude            float64
Venue Longitude           float64
Venue Category             object
dtype: object
shape: (359, 7)


Neighborhood  ...             Venue Category
0          Colaba  ...                Men's Store
1          Colaba  ...          German Restaurant
2          Colaba  ...                        Spa
3          Colaba  ...                Coffee Shop
4          Colaba  ...                    Brewery
5          Colaba  ...          Indian Restaurant
6          Colaba  ...                 Restaurant
7          Colaba  ...         Chinese Restaurant
8          Colaba  ...  Middle Eastern Restaurant
9          Colaba  ...                Pizza Place
10         Colaba  ...                       Café
11         Colaba  ...                      Hotel
12         Colaba  ...         Chinese Restaurant
13  Apollo Bandar  ...                      Diner
14  Apollo Bandar  ...          Indian Restaurant
15  Apollo Bandar  ...               Dessert Shop
16  Apollo Bandar  ...                        Bar
17  Apollo Bandar  ...          Indian Restaurant
18  Apollo Bandar  ...                      Hotel
19  Apollo Bandar  ...                     Lounge
20  Apollo Bandar  ...          Indian Restaurant
21  Apollo Bandar  ...         Falafel Restaurant
22  Apollo Bandar  ...                  Juice Bar
23  Apollo Bandar  ...                Flea Market
24  Apollo Bandar  ...                       Café
25  Apollo Bandar  ...                Coffee Shop
26  Apollo Bandar  ...                       Café
27  Apollo Bandar  ...               Cocktail Bar
28  Apollo Bandar  ...           Asian Restaurant
29  Apollo Bandar  ...                      Hotel

[30 rows x 7 columns]

### No difference. Number of venues did not increase as expected. Let's increase radius to 750.

In [31]:
radius= 750
LIMIT = 100
south_mumbai_venues3 = getNearbyVenues(names=south_mumbai_data['Neighborhood'],
                                   latitudes=south_mumbai_data['Latitude'],
                                   longitudes=south_mumbai_data['Longitude']
                                  )
#print("dtypes:",south_mumbai_venues3.dtypes)
print("shape:",south_mumbai_venues3.shape)
south_mumbai_venues3.head()

Colaba
Apollo Bandar
Fort
Churchgate
Nariman Point
Marine Lines
Walkeshwar
Malabar Hill
Kalbadevi
Bhuleshwar
Masjid Bandar
Darukhana
Pydhoni
Nagpada
Dongri
Byculla
Grand Road
Mazgaon
Mumbai Central
Mahalaxmi
Worli
shape: (359, 7)


Neighborhood  Neighborhood Latitude  ...  Venue Longitude     Venue Category
0       Colaba              18.915091  ...        72.824151        Men's Store
1       Colaba              18.915091  ...        72.827018  German Restaurant
2       Colaba              18.915091  ...        72.823843                Spa
3       Colaba              18.915091  ...        72.824160        Coffee Shop
4       Colaba              18.915091  ...        72.824119            Brewery

[5 rows x 7 columns]

In [0]:
#south_mumbai_venues.to_csv("south_mumbai_venues.csv")

#### 

In [48]:
radius= 500
LIMIT = 200
#Esenyurt 41.0413° N, 28.6941° E ===> 8
#Apollo Bandar	South Mumbai	18.9183748	72.8314429 ==> 46
#Istanbul 41.0082° N, 28.9784° E ==> 100
#London 51.5074° N, 0.1278° W ==> 100
#New York 40.7128° N, 74.0060° W

neighborhood_latitude = 40.7128
neighborhood_longitude = -74.0060

url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
    CLIENT_ID, 
    CLIENT_SECRET, 
    VERSION, 
    neighborhood_latitude, 
    neighborhood_longitude, 
    radius, 
    LIMIT)
url # display URL

# Send the GET request and examine the resutls

results = requests.get(url).json()
#results
#---------------------------------------------------------------
# function that extracts the category of the venue
def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']

#----------------------------------------------------------------
#Now we are ready to clean the json and structure it into a pandas dataframe.

venues = results['response']['groups'][0]['items']
    
nearby_venues = json_normalize(venues) # flatten JSON

# filter columns
filtered_columns = ['venue.name', 'venue.categories', 'venue.location.lat', 'venue.location.lng']
nearby_venues =nearby_venues.loc[:, filtered_columns]

# filter the category for each row
nearby_venues['venue.categories'] = nearby_venues.apply(get_category_type, axis=1)

# clean columns
nearby_venues.columns = [col.split(".")[-1] for col in nearby_venues.columns]

print("url:",url)
print("shape:",nearby_venues.shape)
nearby_venues.to_csv(newyork_venues.csv)
nearby_venues.head()
#-----------------------------------------------------------------



url: https://api.foursquare.com/v2/venues/explore?&client_id=AED1YJAEJO1BSJVJJ5KT2Y2BDGDVPCB21WUXERHTA4B4XXKC&client_secret=5WF0PKUAGWG30ASKZSBV12XHC2IDIQLEHRRPU541XI2JVQCH&v=20180605&ll=40.7128,-74.006&radius=500&limit=200
shape: (100, 4)


/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:44: FutureWarning: pandas.io.json.json_normalize is deprecated, use pandas.json_normalize instead


name       categories        lat        lng
0   The Bar Room at Temple Court        Hotel Bar  40.711448 -74.006802
1  The Beekman, A Thompson Hotel            Hotel  40.711173 -74.006702
2      Alba Dry Cleaner & Tailor  Laundry Service  40.711434 -74.006272
3                 City Hall Park             Park  40.712415 -74.006724
4   Gibney Dance Center Downtown     Dance Studio  40.713923 -74.005661

In [0]:
nearby_venues.to_csv("newyork_venues.csv")

In [60]:
#print('There are {} uniques categories.'.format(len(manhattan_venues['Venue Category'].unique())))
unique_cat = nearby_venues['categories'].unique()
#unique_cat_df = (unique_cat,columns="Uniqe_Category")
unique_cat_df = (unique_cat)
unique_cat_df.head()

#print('There are {} uniques categories.'.format(len(manhattan_venues['Venue Category'].unique())))

AttributeError: ignored

In [63]:
# Creating pandas dataframe from numpy array
# dataset = pd.DataFrame({'Column1': data[:, 0], 'Column2': data[:, 1]})
unique_cat_df = pd.DataFrame({'Uniq_Cat': unique_cat[:]})
unique_cat_df.head()


Uniq_Cat
0        Hotel Bar
1            Hotel
2  Laundry Service
3             Park
4     Dance Studio

In [0]:
dfxxx = dfxxx[['STNAME','CTYNAME']].groupby(['STNAME'])['CTYNAME'] \
                             .count() \
                             .reset_index(name='count') \
                             .sort_values(['count'], ascending=False) \
                             .head(5)

In [75]:
nearby_venues.head()

name       categories        lat        lng
0   The Bar Room at Temple Court        Hotel Bar  40.711448 -74.006802
1  The Beekman, A Thompson Hotel            Hotel  40.711173 -74.006702
2      Alba Dry Cleaner & Tailor  Laundry Service  40.711434 -74.006272
3                 City Hall Park             Park  40.712415 -74.006724
4   Gibney Dance Center Downtown     Dance Studio  40.713923 -74.005661

In [83]:
nearby_venues_grouped = nearby_venues[['categories','lat']].groupby(['categories']).count().reset_index()
nearby_venues_grouped.rename(columns = {'lat':'count'},inplace = True)
nearby_venues_grouped_sorted = nearby_venues_grouped.sort_values(by= 'count',ascending=False)
nearby_venues_grouped_sorted.head(30)


categories  count
17           Coffee Shop      5
32                 Hotel      5
49        Sandwich Place      4
13                  Café      4
25     French Restaurant      3
29                   Gym      3
16          Cocktail Bar      3
15        Clothing Store      3
23    Falafel Restaurant      3
34     Indian Restaurant      2
33             Hotel Bar      2
0    American Restaurant      2
45                  Park      2
46           Pizza Place      2
51                   Spa      2
54      Sushi Restaurant      2
55            Taco Place      2
58             Wine Shop      2
59         Women's Store      2
35    Italian Restaurant      2
30  Gym / Fitness Center      2
60           Yoga Studio      2
11          Burger Joint      2
6                    Bar      2
3             Baby Store      1
7              Bookstore      1
44            Nail Salon      1
9        Bubble Tea Shop      1
8             Boxing Gym      1
47                 Plaza      1

In [82]:
#nearby_venues.shape
nearby_venues.head()
#nearby_venues_grouped = nearby_venues[['categories','lat']].groupby(['categories']).count().reset_index()

nearby_venues_grouped = nearby_venues[['categories','lat']].groupby(['categories']).count().reset_index()
                                                     # .count() \
                                                     # .reset_index(name='count') \
                                                     # .sort_values(['count'],ascending=False) \
nearby_venues_grouped.head(40)   
nearby_venues_grouped.dtypes  
nearby_venues_grouped.columns
nearby_venues_grouped.rename(columns = {'lat':'count'},inplace = True)
nearby_venues_grouped_sorted = nearby_venues_grouped.sort_values(by= 'count',ascending=False)
nearby_venues_grouped_sorted.head()

#df.sort_values(by='col1', ascending=False)
#df.rename(columns = {'Postal code':'PostalCode'}, inplace = True)

categories  count
17        Coffee Shop      5
32              Hotel      5
49     Sandwich Place      4
13               Café      4
25  French Restaurant      3

In [58]:
#df.groupby(["state", "gender"])["last_name"].count()
nearby_venues_grouped = nearby_venues
#nearby_venues_grouped.groupby(["categories"])["name"].count().reset_index()
nearby_venues_grouped = nearby_venues_grouped.groupby(["categories"]).count().reset_index()
nearby_venues_grouped.head(30)
#nearby_venues_grouped=nearby_venues_grouped["categories","name"]
#nearby_venues_grouped.head()
#nearby_venues_grouped_sorted = nearby_venues_grouped
nearby_venues_grouped_sorted = nearby_venues_grouped_sort_values(by="name", ascending=False)
nearby_venues_grouped_sorted.head()
#df.sort_values(by='col1', ascending=False)


NameError: ignored